lib/interpret_and_optimize/helpers/pb_state_management_linker.dart

```diff
@@ -0,0 +1,35 @@
+import 'package:parabeac_core/interpret_and_optimize/entities/subclasses/pb_intermediate_node.dart';
+import 'package:parabeac_core/interpret_and_optimize/state_management/intermediate_state.dart';
+import 'package:parabeac_core/interpret_and_optimize/state_management/intermediate_variation.dart';
+
+class PBStateManagementLinker {
```

Comment:
The linker keeps track of the variations in State Management (SM) configuration. When giving it a PBIntermediateNode that should be interpreted as part of SM configuration, two things can happen:

1.  The node is a default node (for now, the first node to be registered):
        - Add it to the map so we can add to its state graph

2. The node is not a default node:
        - Add it to the default node's state graph